compiler/qsc_rir/src/passes/ssa_transform.rs

```diff
@@ -0,0 +1,343 @@
+            Instruction::Shl(lhs, rhs, out) => {
+                block.0.push(Instruction::Shl(
+                    lhs.mapped(var_map),
+                    rhs.mapped(var_map),
+                    out,
+                ));
+            }
+            Instruction::Ashr(lhs, rhs, out) => {
+                block.0.push(Instruction::Ashr(
+                    lhs.mapped(var_map),
+                    rhs.mapped(var_map),
+                    out,
+                ));
+            }
+            Instruction::Icmp(cond, lhs, rhs, out) => {
+                block.0.push(Instruction::Icmp(
+                    cond,
+                    lhs.mapped(var_map),
+                    rhs.mapped(var_map),
+                    out,
+                ));
+            }
+            Instruction::LogicalNot(operand, out) => {
+                block
+                    .0
+                    .push(Instruction::LogicalNot(operand.mapped(var_map), out));
+            }
+            Instruction::LogicalAnd(lhs, rhs, out) => {
+                block.0.push(Instruction::LogicalAnd(
+                    lhs.mapped(var_map),
+                    rhs.mapped(var_map),
+                    out,
+                ));
+            }
+            Instruction::LogicalOr(lhs, rhs, out) => {
+                block.0.push(Instruction::LogicalOr(
+                    lhs.mapped(var_map),
+                    rhs.mapped(var_map),
+                    out,
+                ));
+            }
+            Instruction::BitwiseNot(operand, out) => {
+                block
+                    .0
+                    .push(Instruction::BitwiseNot(operand.mapped(var_map), out));
+            }
+            Instruction::BitwiseAnd(lhs, rhs, out) => {
+                block.0.push(Instruction::BitwiseAnd(
+                    lhs.mapped(var_map),
+                    rhs.mapped(var_map),
+                    out,
+                ));
+            }
+            Instruction::BitwiseOr(lhs, rhs, out) => {
+                block.0.push(Instruction::BitwiseOr(
+                    lhs.mapped(var_map),
+                    rhs.mapped(var_map),
+                    out,
+                ));
+            }
+            Instruction::BitwiseXor(lhs, rhs, out) => {
+                block.0.push(Instruction::BitwiseXor(
+                    lhs.mapped(var_map),
+                    rhs.mapped(var_map),
+                    out,
+                ));
+            }
+
+            // Phi nodes are handled separately in the SSA transformation, but need to be passed through.
+            Instruction::Phi(..) | Instruction::Jump(..) | Instruction::Return => {
+                block.0.push(instr);
+            }
+        }
+    }
+}
+
+impl Operand {
+    fn mapped(&self, var_map: &impl VariableMapper) -> Operand {
+        match self {
+            Operand::Literal(_) => *self,
+            Operand::Variable(var) => var_map.to_operand(*var),
+        }
+    }
+
+    fn get_type(&self) -> Ty {
```

Comment:
nit: this is a generic method that would make sense to make public and implement it in rir.rs directly.

Reply:
Moved into RIR itself.